REST_based_Auth_API/my_rest_api/core/views.py

```diff
@@ -0,0 +1,371 @@
+    serializer_class = UserSerializer
+    permission_classes = (IsAuthenticated,)
+    queryset = User.objects.all()
+
+
+class PostView(viewsets.ModelViewSet):
+    serializer_class = PostSerializer
+    permission_classes = (IsAuthenticated,)
+    queryset = Post.objects.all()
+
+    def get_queryset(self):
+        user = self.request.user
+        return self.queryset.filter(user=user)
+
+
+class CommentView(viewsets.ModelViewSet):
+    serializer_class = CommentSerializer
+    permission_classes = (IsAuthenticated,)
+    queryset = Comment.objects.all()
+
+    def get_queryset(self):
+        post_id = self.request.data["post"]
+        comment_id = self.request.data["comment"]
+
+        if comment_id:  # checking it because it can be null
+            return self.queryset.filter(comment=comment_id)
+        return self.queryset.filter(post=post_id)
+
+    def create(self, request, *args, **kwargs):
+        signed_in_user = self.request.user.id
+        post_id = self.request.data["post"]
+        comment_id = self.request.data["comment"]
+
+        commentor = User.objects.get(pk=signed_in_user)
+        parent_post = Post.objects.get(pk=post_id)
+        if not parent_post:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+        post_author = parent_post.user_id
+
+        if comment_id:
+            parent_comment = Comment.objects.get(pk=comment_id)
+            if not parent_comment:
+                return Response(status=status.HTTP_400_BAD_REQUEST)
+            comment_type = Comment.CommentType.COMMENT
+            notification_text = (
+                f"{commentor.username} has replied to a Comment on your post"
+            )
+
+        else:
+            comment_type = Comment.CommentType.POST
+            notification_text = f"{commentor.username} has Commented on your post"
+
+        serializer = self.serializer_class(data=self.request.data)
+        serializer.is_valid(raise_exception=True)
+
+        create_notification(
+            request,
+            to_user=User(post_author),
+            notification=notification_text,
+            notification_source_type=comment_type,
+            user_by=signed_in_user,
+            notification_source_id=notification_source_id,
+        )
+        create_notification(
+            request,
+            to_user=User(post_author),
+            notification=notification_text,
+            notification_source_type=comment_type,
+            user_by=User(signed_in_user),
+            post=post_id,
+            comment=comment_id,
+        )
+        comment = serializer.save()
+        return Response(self.serializer_class(comment).data)
+
+
+class RequestView(viewsets.ModelViewSet):
+    serializer_class = RequestSerializer
+    permission_classes = (IsAuthenticated,)
+    queryset = Request.objects.all()
+
+    def get_queryset(self):
+        signed_in_user = self.request.user.id
+        return Request.objects.filter(requestee=signed_in_user, status="P")
```

Comment:
_Sigh_... I told you many times not to use raw enum values directly like this. It should be `Request.Status.PENDING`.